File: eng/docs/agent_skills/csharp_typing/examples/integration_tests/ProtocolPathTest.cs
// ============================================================
// Integration test: Protocol / path
// Family: protocol_path
// Level: cmp_int (component integration — event ordering)
// ============================================================
//
// Verifies that on the happy path the execution engine emits
// observability events in the correct prescribed order.

using FluentAssertions;
using Xunit;

// Adjust these usings to your real namespace.

namespace Acme.ExecutionEngine.Tests.Integration;

// ============================================================
// Domain types (self-contained example)
// ============================================================

internal sealed record ProtocolExecutionOutcome(string Status, string Reason);

// ============================================================
// Event capture implementing observability port
// ============================================================

internal sealed class ProtocolEventCapture
{
    private readonly List<Dictionary<string, object>> _events = [];

    public void Emit(Dictionary<string, object> ev) => _events.Add(ev);

    public List<string> EventNames =>
        _events.Select(e => (string)e["ev"]).ToList();
}

// ============================================================
// Collaborator: successful agent store
// ============================================================

internal sealed class SuccessfulAgentStore
{
    public Dictionary<string, object> ResolveAgent(string agentId) =>
        new() { { "agent_id", agentId }, { "provider", "local" } };
}

// ============================================================
// Engine under test
// ============================================================

internal sealed class ProtocolExecutionEngine
{
    private readonly SuccessfulAgentStore _agentStore;
    private readonly ProtocolEventCapture _obs;

    public ProtocolExecutionEngine(
        SuccessfulAgentStore agentStore,
        ProtocolEventCapture obs)
    {
        _agentStore = agentStore;
        _obs = obs;
    }

    public ProtocolExecutionOutcome Execute(string agentId)
    {
        _obs.Emit(new() { { "ev", "EXECUTION_STARTED" } });
        var cfg = _agentStore.ResolveAgent(agentId);
        _obs.Emit(new() { { "ev", "AGENT_RESOLVED" }, { "provider", cfg["provider"] } });
        _obs.Emit(new() { { "ev", "EXECUTION_DONE" } });
        return new ProtocolExecutionOutcome(Status: "DONE", Reason: "SUCCESS");
    }
}

// ============================================================
// Tests
// ============================================================

[Trait("Level", "cmp_int")]
[Trait("Family", "protocol_path")]
public sealed class ProtocolPathTest
{
    [Fact]
    public void HappyPathEmitsExpectedOrderedEvents()
    {
        // Arrange
        var obs = new ProtocolEventCapture();
        var engine = new ProtocolExecutionEngine(
            new SuccessfulAgentStore(), obs);

        // Act
        var result = engine.Execute("agent_a");

        // Assert — correct result
        result.Should().Be(
            new ProtocolExecutionOutcome(Status: "DONE", Reason: "SUCCESS"));

        // Assert — events in correct order
        obs.EventNames.Should().BeEquivalentTo(
            new[] { "EXECUTION_STARTED", "AGENT_RESOLVED", "EXECUTION_DONE" },
            options => options.WithStrictOrdering());
    }
}

File: eng/docs/agent_skills/python_typing/examples/integration tests/protocol_path_test.py
from __future__ import annotations

from dataclasses import dataclass

import pytest

from dictionary.implementation.TL_CORE import Result


@dataclass(frozen=True, slots=True)
class ExecutionOutcome:
    status: str
    reason: str


class ObservabilityPort:
    def emit(self, event: dict) -> None:
        raise NotImplementedError


class EventCapture(ObservabilityPort):
    def __init__(self) -> None:
        self.events: list[dict] = []

    def emit(self, event: dict) -> None:
        self.events.append(event)


class AgentStorePort:
    def resolve_agent(self, agent_id: str) -> dict:
        raise NotImplementedError


class SuccessfulAgentStore(AgentStorePort):
    def resolve_agent(self, agent_id: str) -> dict:
        return {"agent_id": agent_id, "provider": "local"}


class ExecutionEngine:
    def __init__(self, agent_store: AgentStorePort, obs: ObservabilityPort) -> None:
        self._agent_store = agent_store
        self._obs = obs

    def execute(self, agent_id: str) -> Result[ExecutionOutcome]:
        self._obs.emit({"ev": "EXECUTION_STARTED"})
        cfg = self._agent_store.resolve_agent(agent_id)
        self._obs.emit({"ev": "AGENT_RESOLVED", "provider": cfg["provider"]})
        self._obs.emit({"ev": "EXECUTION_DONE"})
        return Result(
            outcome="ok",
            ok=ExecutionOutcome(status="DONE", reason="SUCCESS"),
        )


@pytest.fixture
def protocol_slice() -> tuple[ExecutionEngine, EventCapture]:
    obs = EventCapture()
    ee = ExecutionEngine(agent_store=SuccessfulAgentStore(), obs=obs)
    return ee, obs


@pytest.mark.cmp_int
@pytest.mark.protocol_path
def test_execution_happy_path_emits_expected_ordered_events(
    protocol_slice: tuple[ExecutionEngine, EventCapture],
) -> None:
    ee, obs = protocol_slice

    result = ee.execute("agent_a")

    assert result.outcome == "ok"
    assert result.ok == ExecutionOutcome(status="DONE", reason="SUCCESS")
    assert [e["ev"] for e in obs.events] == [
        "EXECUTION_STARTED",
        "AGENT_RESOLVED",
        "EXECUTION_DONE",
    ]
